src/huggingface_hub/utils/_hf_folder.py

```diff
@@ -39,19 +48,41 @@ def get_token(cls) -> Optional[str]:
         """
         Get token or None if not existent.
 
-        Note that a token can be also provided using the
-        `HUGGING_FACE_HUB_TOKEN` environment variable.
+        Note that a token can be also provided using the `HUGGING_FACE_HUB_TOKEN` environment variable.
+
+        Token is saved in the huggingface home folder. You can configure it by setting
+        the `HF_HOME` environment variable. Previous location was `~/.huggingface/token`.
+        If token is not found in the new location, previous location is read. If it exists,
+        token is duplicated to the new location.
+        For more details, see https://github.com/huggingface/huggingface_hub/issues/1232.
 
         Returns:
             `str` or `None`: The token, `None` if it doesn't exist.
         """
+        cls._warn_on_old_token_path()
```

Comment:
I would just warn once, the first time we re-write the token in the new place, instead of every time. There will be an old token living in the old path if the user forgets about the warning but I don't think it causes any security threat (given the new path is kind of the same)?

Reply:
Yep, it's better that way. I have adapted the code so that warning is issued only once when token is copied from old to new path. Shouldn't pose any problem. Thanks for the suggestion